DuckDuckGo/Updates/UpdateController.swift

```diff
@@ -249,12 +259,18 @@ extension UpdateController: SPUUpdaterDelegate {
 
     func updaterDidNotFindUpdate(_ updater: SPUUpdater, error: any Error) {
         let nsError = error as NSError
-        guard let item = nsError.userInfo["SULatestAppcastItemFound"] as? SUAppcastItem else { return }
+        guard let item = nsError.userInfo[SPULatestAppcastItemFoundKey] as? SUAppcastItem else { return }
 
         Logger.updates.log("Updater did not find update: \(String(describing: item.displayVersionString))(\(String(describing: item.versionString)))")
         PixelKit.fire(DebugEvent(GeneralPixel.updaterDidNotFindUpdate, error: error))
 
-        cachedUpdateResult = UpdateCheckResult(item: item, isInstalled: true)
+        // Edge case: User upgrades to latest version within their rollout group
+        // But fetched release notes are outdated due to rollout group reset
+        let needsLatestReleaseNote = {
+            guard let reason = nsError.userInfo[SPUNoUpdateFoundReasonKey] as? Int else { return false }
+            return reason == Int(Sparkle.SPUNoUpdateFoundReason.onNewerThanLatestVersion.rawValue)
+        }()
+        cachedUpdateResult = UpdateCheckResult(item: item, isInstalled: true, needsLatestReleaseNote: needsLatestReleaseNote)
```

Comment:
Not great, but the user has 1/7 chance of seeing the mismatched release note, so I'm adding this workaround till we address it properly.